vpr/src/place/place_delay_model.h

```diff
@@ -95,9 +109,14 @@ class OverrideDelayModel : public PlaceDelayModel {
         short delta_x;
         short delta_y;
 
-        friend bool operator<(const t_override& lhs, const t_override& rhs) {
-            return std::tie(lhs.from_type, lhs.to_type, lhs.from_class, lhs.to_class, lhs.delta_x, lhs.delta_y)
-                   < std::tie(rhs.from_type, rhs.to_type, rhs.from_class, rhs.to_class, rhs.delta_x, rhs.delta_y);
+        //A combination of ALWAYS_INLINE attribute and std::lexicographical_compare
+        //is required for operator< to be inlined by compiler.
+        //Proper inlining of the function reduces place time by around 5%.
+        //For more information: https://github.com/verilog-to-routing/vtr-verilog-to-routing/issues/1225
+        friend ALWAYS_INLINE bool operator<(const t_override& lhs, const t_override& rhs) {
+            const short* left = reinterpret_cast<const short*>(&lhs);
+            const short* right = reinterpret_cast<const short*>(&rhs);
+            return std::lexicographical_compare(left, left + 6, right, right + 6);
```

Comment:
I think we also need to explicitly state what assumptions this code is making. In particular, I think it is assuming:

- All data members of t_override are shorts
- There is no padding in t_override
- Based on the above, treating t_override's memory layout as an array of shorts is the same as a member-wise comparison

We should also encode these assumptions into some compile-time checks, otherwise we are setting ourselves up for some very challenging to find bugs in the future (when t_override changes).

I'd suggest something like:
```cpp
static_assert(sizeof(t_override) = sizeof(t_override::from_type)
                                   + sizeof(t_override::to_type)
                                    + sizeof(t_override::from_class)
                                    + sizeof(t_override::to_class)
                                    + sizeof(t_override::delta_x)
                                    + sizeof(t_override::delta_y),
              "Expect t_override to have a memory layout equivalent to an array of short (no padding)");
static_assert(sizeof(t_override::from_type) == sizeof(short), "Expect all t_override data members to be shorts");
static_assert(sizeof(t_override::to_type) == sizeof(short), "Expect all t_override data members to be shorts");
static_assert(sizeof(t_override::from_class) == sizeof(short), "Expect all t_override data members to be shorts");
static_assert(sizeof(t_override::to_class) == sizeof(short), "Expect all t_override data members to be shorts");
static_assert(sizeof(t_override::delta_x) == sizeof(short), "Expect all t_override data members to be shorts");
static_assert(sizeof(t_override::delta_y) == sizeof(short), "Expect all t_override data members to be shorts");
```
which would verify there is no padding in the t_override memory layout and that each data member is a short. This would catch potential bugs caused by changing the number of data members in t_override, or the data type of members in t_override.

We could then also modify the comparison to not be explicitly hard coded for 6 data members:
```cpp
constexpr size_t NUM_T_OVERRIDE_MEMBERS = sizeof(t_override) / sizeof(short);
return std::lexicographical_compare(left, left + NUM_T_OVERRIDE_MEMBERS, right, right + NUM_T_OVERRIDE_MEMBERS);
```